Only in ../CS214-SPRING2019/Asst3/libtar/: .DS_Store
File ./.git is a directory while file ../CS214-SPRING2019/Asst3/libtar/.git is a regular file
Only in ./compat: fnmatch.c
Only in ./compat: gethostbyname_r.c
Only in ./compat: gethostname.c
Only in ./compat: getservbyname_r.c
Only in ./compat: glob.c
Only in ./compat: inet_aton.c
Only in ./compat: snprintf.c
Only in ./compat: strdup.c
Only in ./compat: strlcat.c
Only in ./compat: strrstr.c
Only in ./compat: strsep.c
Only in ../CS214-SPRING2019/Asst3/libtar/: config.h
diff -r ./lib/decode.c ../CS214-SPRING2019/Asst3/libtar/lib/decode.c
29c29
< 	char filename[MAXPATHLEN];
---
>     static __thread char filename[MAXPATHLEN];
38c38
< 		return strdup(filename);
---
> 		return filename;
42c42
< 	return strdup(filename);
---
> 	return filename;
diff -r ./lib/encode.c ../CS214-SPRING2019/Asst3/libtar/lib/encode.c
32,33c32,35
< 	if (t->options & TAR_GNU)
< 		strncpy(t->th_buf.magic, "ustar  ", 8);
---
> 	if (t->options & TAR_GNU) {
>         strncpy(t->th_buf.magic, "ustar ", 6);
>         strncpy(t->th_buf.version, " ", 2);
>     }
124a127,136
> 	    /*
> 	     *  MEMORY LEAK FIXED:
> 	     *  There would be a memory leak in gnu_longlink if we use set_link for two times consecutively and both times we have a GNU longlink
> 	     *  Need to free the previous one before duplicating a new one
> 	     *  The last copy of the gnu_longlink will be freed at tar_close
> 	     */
> 	    if (t->th_buf.gnu_longlink) {
> 	        free(t->th_buf.gnu_longlink);   // free previous link before duplicating a new one
> 	    }
> 
136c148
< 		t->th_buf.gnu_longlink = NULL;
---
> 		t->th_buf.gnu_longlink = NULL;          // reset this to NULL to notify th_write we are not using long link
diff -r ./lib/handle.c ../CS214-SPRING2019/Asst3/libtar/lib/handle.c
123a124,136
> 
> 	/*
> 	 * MEMORY LEAK FIXED:
> 	 *
> 	 * Even though th_set_path and th_set_link(AFTER PATCH) free the previous pointer before allocate a new one
> 	 * The last pointer never got freed, so when we are freeing t, we also need to free those.
> 	 */
> 
> 	if (t->th_buf.gnu_longlink)
> 	    free(t->th_buf.gnu_longlink);
> 	if (t->th_buf.gnu_longname)
> 	    free(t->th_buf.gnu_longname);
> 
diff -r ./lib/libtar.h ../CS214-SPRING2019/Asst3/libtar/lib/libtar.h
21a22,23
> #include <string.h>
> 
Only in ../CS214-SPRING2019/Asst3/libtar/listhash: libtar_hash.c
Only in ../CS214-SPRING2019/Asst3/libtar/listhash: libtar_list.c
Only in ../CS214-SPRING2019/Asst3/libtar/listhash: libtar_listhash.h
